src/Monolog/Processor/WebProcessor.php

```diff
@@ -44,23 +44,21 @@ class WebProcessor
      */
     public function __construct($serverData = null, array $extraFields = null)
     {
-        if (null === $serverData) {
+        if ($serverData === null || count($serverData) === 0) {
```

Comment:
IMO, the previous signature is better:

- passing `null` means that Monolog will use $_SERVER
- passing an array or ArrayAccess object means it will be used

Now, you add a magic case where passing an empty array does not behave like passing a non-empty array